packages/components/src/lib/d3/index.ts

```diff
@@ -28,3 +28,48 @@ export function sqScaleToD3(
       throw new Error(`Unknown scale: ${scale satisfies never}`);
   }
 }
+
+/**
+ * A function to adjust the height of PDF values in accordance with non-linear scales.
+ * This is achieved by multiplying the values by the inverse of the derivative of the scale function at each point.
+ * The adjustment allows the PDF values to be more representative of the underlying data,
+ * providing a better fit for a histogram representation of the data with the given scales.
+ */
+function pdfScaleHeightAdjustment(
+  scale: SqScale
+): (x: number, y: number) => number {
+  switch (scale.tag) {
+    case "linear":
+      return (_, y) => y;
+    case "symlog":
+    case "log":
+      // Technically, we should also muliply by the log of the base of the log scale.
+      // However, this is a constant, and we don't show the y-axis anyway.
+      // Also, the value for symlog should be slightly different from log, but we ignore that for now.
```

Comment:
Symlog case seem important to handle.

My calculus skills are rusty, but if I trust ChatGPT, `1 / derivative(symlog function)` is `abs(x) + c`.

Reply:
Then the difference is the ``+c`` part. This seems weird to me as a think we'd add - it would add probability mass absolutely everywhere?

Reply:
I assume C is the constant? And the full equation would be, 
```
return y * (this._value.constant + Math.abs(x))
```

If that's the case, I'd expect this to have a small effect, as Math.abs(x) >> this.value.constant for most values, unless this.value.constant were really high.

Reply:
> If that's the case, I'd expect this to have a small effect, as Math.abs(x) >> this.value.constant for most values, unless this.value.constant were really high.

Doesn't that depends on the distribution, and we can't assume that apriori? User probably won't force symlog dist on something like `-1 to 1` intentionally, but for example in relative values `ListView` we construct `Plot.dist` with symlog scales by default, and the range of values there could be mostly in that range, in many cases.

Reply:
Not saying we shouldn't do it, but I'd be surprised if over 5% of of dists had 80% mass less than abs(1).

Reply:
> over 5% of of dists had 80% mass less than abs(1)

In relative values, every A to B has an opposite B to A, so something like "50% of dists have >50% mass less than abs(1)" must be true.

I mean these views:
<img width="490" alt="Captura de pantalla 2023-07-25 a la(s) 12 12 58" src="https://github.com/quantified-uncertainty/squiggle/assets/89368/b772cd40-84dd-45f8-ba52-1b16bab5ecbb">

Reply:
I struggled a bit to find an example where this helps, but it seems fairly clear for this code:
```
Plot.dist({
    dist: uniform(1,10),
    xScale: Scale.symlog({base:100000000}),
    showSummary: true
})
```

<img width="1176" alt="image" src="https://github.com/quantified-uncertainty/squiggle/assets/377065/d42d031a-3bee-42ab-87d1-1f88174e62b5">

Vs. 

<img width="1176" alt="image" src="https://github.com/quantified-uncertainty/squiggle/assets/377065/df6d34b0-97f5-47ae-9952-51ac0faa1bc2">

I'll take it, thanks!